examples/ts/nextjs/pages/api/bad.ts

```diff
@@ -0,0 +1,36 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import '../../lib/bugsnag';
+import Bugsnag from '@bugsnag/js';
+
+var middleware = Bugsnag.getPlugin('express')
+
+// Helper method to wait for a middleware to execute before continuing
+// And to throw an error when an error happens in a middleware
+function runMiddleware(req, res, fn) {
+  return new Promise((resolve, reject) => {
+    fn(req, res, (result) => {
+      if (result instanceof Error) {
+        return reject(result)
+      }
+
+      return resolve(result)
+    })
+  })
+}
+
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
```

Comment:
This is an example of wrapping a next.js API route with Bugsnag. It's fairly involved but I'm not sure there are many alternatives as next.js doesn't seem to provide error handling here so you need to catch your own exceptions.